File: 0202-happy-number/0202-happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = {n}

        while True:
            res = 0
            while n:
                res += (n % 10) ** 2
                n //= 10
            if res == 1: return True
            if res in seen: return False

            n = res
            seen.add(n)

File: 0202-happy-number/0202-happy-number.java
class Solution {
    public boolean isHappy(int n) {
        Set<Integer> seen = new HashSet<>();

        while (true) {
            if (seen.contains(n)) {
                return false;
            }
            if (n == 1) {
                return true;
            }
            seen.add(n);
            int temp = 0;

            while (n != 0) {
                temp += (n % 10) * (n % 10);
                n /= 10;
            }
            n = temp;
        }
    }
}